packages/back-end/src/validators/metric-time-series.ts

```diff
@@ -58,7 +58,7 @@ export const metricTimeSeriesSchema = z
     dateUpdated: z.date(),
 
     metricId: z.string(),
-    source: z.enum(["experiment"]),
+    source: z.enum(["experiment", "safe-rollout"]),
```

Comment:
Just checking: Is it a problem that sometimes in safe rollouts one side or the other of the CI is always Infinity? I see in the datatype for the CI it says the values must be a number. Will this work if we have one end of the CI is infinity? I think it should, but worth testing.

Reply:
Good point! Zod does treat `Infinity` as a valid `z.number()`, but after parsing it actually becomes `null` instead of keeping it as Infinity 🤔 (playground).

I'll think more about the implications of this and what we might need to update to account for that.

Reply:
Ohh my understanding was wrong. Zod does properly treat `Infinity` as a valid number and accepts it, and it is properly stored in our Database.

The playground though (and our API) show the result parsed into JSON and the JSON schema does not accept Infinity as a valid number value, therefore it gets converted to `null`.

So while this is indeed a problem, it is in a different layer (fetching the data from the client) instead of Zod validation.

Reply:
And confirmed we are good because we updated the types to be `null | number` to account for this and handle it by falling back to `Infinity` ([example](https://github.com/growthbook/growthbook/blob/main/packages/front-end/components/Experiment/ResultsTableTooltip/ResultsTableTooltip.tsx#L232)).

Reply:
Very annoying that we can't pass Infinity back and forth